File: engine_java_interface/test/src/KeyEvent.java
public final class KeyEvent {
    public KeyEvent( KeyEventType type, int key, double timePressed ){
        this.type = type;
        this.key = key;
        this.timePressed = timePressed;
    }

    public final KeyEventType type;
    public final int key;
    public final double timePressed;
}

File: engine/libs/mcgl-engine-core/src/Eventing/KeyEvent.cpp
#include "Eventing/KeyEvent.h"


namespace eventing {


const char* to_string( KeyEventType type ) {
	switch ( type ) {
		case KeyEventType::Pressed:
			return "Pressed";
		case KeyEventType::Down:
			return "Down";
		case KeyEventType::Released:
			return "Released";
		default:
			return "<undefined>";
	}
}


}